File: my-folder/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hashy = new HashMap<Integer, Integer>();
        
        for(int i = 0; i < nums.length; i++){
            hashy.put(nums[i], i);
        }
        
        for(int i = 0; i < nums.length; i++){
            int find = target - nums[i];
            if(hashy.containsKey(find) && hashy.get(find) != i){
                return new int[]{i, hashy.get(find)};
            }
        }
        return new int[0];
    }
}

File: my-folder/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        vis = {}
        
        for i, n in enumerate(nums):
            if n in vis and n + n == target:
                return [vis[n], i]
            vis[n] = i
        
        for i, n in enumerate(nums):
            if target - n != n and target - n in vis:
                return (i, vis[target - n])
        
        return [-1, -1]
